services/server/src/server/services/utils/database-util.ts

```diff
@@ -163,8 +169,210 @@ export type GetSourcifyMatchByChainAddressResult = Tables.SourcifyMatch &
+  block_number: "contract_deployments.block_number",
+  transaction_index: "contract_deployments.transaction_index",
+  deployer:
+    "concat('0x', encode(contract_deployments.deployer, 'hex')) as deployer",
+  sources: `${sourcesAggregation} as sources`,
+  language: "INITCAP(compiled_contracts.language) as language",
+  compiler: "compiled_contracts.compiler",
+  version: "compiled_contracts.version as version",
+  compiler_settings: "compiled_contracts.compiler_settings",
+  name: "compiled_contracts.name",
+  fully_qualified_name: "compiled_contracts.fully_qualified_name",
+  abi: "compiled_contracts.compilation_artifacts->'abi' as abi",
+  metadata: "sourcify_matches.metadata",
+  storage_layout:
+    "compiled_contracts.compilation_artifacts->'storageLayout' as storage_layout",
+  userdoc: "compiled_contracts.compilation_artifacts->'userdoc' as userdoc",
+  devdoc: "compiled_contracts.compilation_artifacts->'devdoc' as devdoc",
+  std_json_input: `json_build_object(
+    'language', INITCAP(compiled_contracts.language), 
+    'sources', ${sourcesAggregation},
+    'settings', compiled_contracts.compiler_settings
+  ) as std_json_input`,
+  std_json_output: `json_build_object(
+    'sources', compiled_contracts.compilation_artifacts->'sources',
+    'contracts', json_build_object(
+      substring(
+        compiled_contracts.fully_qualified_name, 
+        1, 
+        length(compiled_contracts.fully_qualified_name) - length(split_part(compiled_contracts.fully_qualified_name, ':', -1)) - 1
+      ), 
+      json_build_object(
+        split_part(compiled_contracts.fully_qualified_name, ':', -1), json_build_object(
+          'abi', compiled_contracts.compilation_artifacts->'abi',
+          'metadata', cast(sourcify_matches.metadata as text),
+          'userdoc', compiled_contracts.compilation_artifacts->'userdoc',
+          'devdoc', compiled_contracts.compilation_artifacts->'devdoc',
+          'storageLayout', compiled_contracts.compilation_artifacts->'storageLayout',
+          'evm', json_build_object(
+            'bytecode', json_build_object(
+              'object', concat('0x', encode(recompiled_creation_code.code, 'hex')),
+              'sourceMap', compiled_contracts.creation_code_artifacts->'sourceMap',
+              'linkReferences', compiled_contracts.creation_code_artifacts->'linkReferences'
+            ),
+            'deployedBytecode', json_build_object(
+              'object', concat('0x', encode(recompiled_runtime_code.code, 'hex')),
+              'sourceMap', compiled_contracts.runtime_code_artifacts->'sourceMap',
+              'linkReferences', compiled_contracts.runtime_code_artifacts->'linkReferences',
+              'immutableReferences', compiled_contracts.runtime_code_artifacts->'immutableReferences'
+            )
+          )
+        )
+      )
+    )
+  ) as std_json_output`,
+};
+
+export type SourcifyMatchDbProperties =
```

Comment:
This type name is a bit confusing. SourcifyMatch is also a table. I wonder if we can find a better name:

How `SourcifyMapDbProperties = SOURCIFY_MAP_DB_PROPERTIES_TO_SELECTORS`  sounds like?

Reply:
Oh yes that's indeed quite ambigous.

Reply:
How's a simple `STORED_PROPERTIES_TO_SELECTORS`?

Reply:
Perfect! :)